src/journal-remote/journal-remote-main.c

```diff
@@ -175,7 +180,25 @@ static int request_meta(void **connection_cls, int fd, char *hostname, Compressi
 
         log_debug("Added RemoteSource as connection metadata %p", source);
 
-        source->compression = compression;
+        for (int i = 0; arg_compression != NULL && arg_compression[i] != NULL; i++)
+                count++;
```

Comment:
you are counting with two iterator counters to the same level. get rid of `i` and just count directly with `count`